source/appModules/foobar2000.py

```diff
@@ -29,28 +45,47 @@ def event_gainFocus(self, obj, nextHandler):
 
 	def getElapsedAndTotal(self):
 		if not self.statusBar: return None
-		text = self.statusBar.firstChild.name
+		statusBarContents = self.statusBar.firstChild.name
 		try:
-			ltime = text.split("|")[4].split(" / ")
+			playingTimes = statusBarContents.split("|")[4].split("/")
+			return playingTimes[0], playingTimes[1]
 		except IndexError:
 			return None
-		elapsedTime = calendar.timegm(time.strptime(ltime[0].strip(),getFormat(ltime[0])))
-		totalTime = calendar.timegm(time.strptime(ltime[1].strip(),getFormat(ltime[1])))
-		return elapsedTime,totalTime
 
-	def script_reportRemainingTime(self,gesture):
-		times=self.getElapsedAndTotal()
-		if times is None:
+	def getElapsedAndTotalIfPlaying(self):
+		times = self.getElapsedAndTotal()
+		if not times:
 			# Translators: Reported when no track is playing in Foobar 2000.
 			ui.message(_("No track playing"))
-			return
-		elapsedTime,totalTime = times
-		remainingTime = totalTime-elapsedTime
-		msg = time.strftime(getOutputFormat(remainingTime),time.gmtime(remainingTime))
-		ui.message(msg)
-	# Translators: The description of an NVDA command for Foobar 2000.
+		return times
+
+	def script_reportRemainingTime(self,gesture):
+		elapsedTime, totalTime = self.getElapsedAndTotalIfPlaying()
+		if elapsedTime is not None and totalTime is not None:
+			parsedElapsedTime = parseIntervalToTimestamp(elapsedTime)
+			parsedTotalTime = parseIntervalToTimestamp(totalTime)
+			remainingTime = parsedTotalTime - parsedElapsedTime
+			msg = time.strftime(getOutputFormat(remainingTime), time.gmtime(remainingTime))
+			ui.message(msg)
+	# Translators: The description of an NVDA command for reading the remaining time of the currently playing track in Foobar 2000.
 	script_reportRemainingTime.__doc__ = _("Reports the remaining time of the currently playing track, if any")
 
+	def script_reportElapsedTime(self,gesture):
+		elapsedTime, _ = self.getElapsedAndTotalIfPlaying()

```

Comment:
In a fp world, I like this, but gettext is _ in NVDA, and I can see a dev going to use _ later in this function and getting a nasty surprise.